src/common/log/log.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+/**
+ * Logging utility class.
+ */
+
+import {CommandStatus} from "../commandExecutor";
+import {LogHelper, LogChannelType, LogLevel} from "./logHelper";
+import {ILogger, OutputChannelLogger, StreamLogger, ConsoleLogger} from "./loggers";
+import {OutputChannel} from "vscode";
+
+export module Log {
+    let globalLogger: ILogger;
+    /**
+     * Logs a message.
+     */
+    export function CreateGlobalLogger(targetChannel?: any) {
+        switch (LogHelper.getLogChannelType(targetChannel)) {
+            case LogChannelType.OutputChannel:
+                globalLogger = new OutputChannelLogger(targetChannel);
+                break;
+
+            case LogChannelType.WritableStream:
+                globalLogger = new StreamLogger(targetChannel);
+                break;
+
+            case LogChannelType.Console:
+            default:
+                globalLogger = new ConsoleLogger();
+                break;
+        }
+    }
+
+    /**
+     * Logs a message.
+     */
+    export function logMessage(message: string, formatMessage: boolean = true) {
+        globalLogger ? globalLogger.logMessage(message, formatMessage) : new ConsoleLogger().logMessage(message, formatMessage);
```

Comment:
Instead of using this if we should just initialize globalLogger in line 14 to new ConsoleLogger(), and let people overwrite it with a call to CreateGlobalLogger if they want to.